http/handler.go

```diff
@@ -27,6 +27,9 @@ const (
 	// not to use request forwarding
 	NoRequestForwardingHeaderName = "X-Vault-No-Request-Forwarding"
 
+	// NoCORS is the name of the header telling Vault not to use CORS handler.
+	NoCORS = "X-Vault-No-CORS"
```

Comment:
I put this in for the CLI client. CORS only matters to the browser. So rather than putting in a check for specific user agents I put a header with a weird value on requests that came in from the CLI. 

Certainly someone could dig through the code and find that header and put it in to override the admin's CORS settings, however, the browser will still require CORS headers from the server for the request to go through.